src/main/java/de/tum/in/www1/artemis/config/migration/entries/MigrationEntry20230808_203400.java

```diff
@@ -0,0 +1,460 @@
+        }
+    }
+
+    /**
+     * Migrates a single test build plan.
+     *
+     * @param exercise The exercise to migrate
+     * @param errorMap The error map to write errors to
+     */
+    private void migrateTestBuildPlan(ProgrammingExercise exercise, Map<ProgrammingExercise, Boolean> errorMap) {
+        if (exercise.getTestRepositoryUrl() == null) {
+            /*
+             * when the test repository url is null, we don't have to migrate the test build plan, saving multiple API calls
+             * this is also only needed for Jenkins, as Bamboo does not have a separate test build plan
+             */
+            return;
+        }
+        try {
+            ciMigrationService.orElseThrow().deleteBuildTriggers(null, null, exercise.getVcsTestRepositoryUrl());
+        }
+        catch (VersionControlException | ContinuousIntegrationException | NoSuchElementException | RestClientException e) {
+            log.warn("Failed to delete test build triggers for exercise {} with test repository {}", exercise.getId(), exercise.getVcsTestRepositoryUrl(), e);
+            errorMap.put(exercise, true);
+        }
+    }
+
+    /**
+     * Migrates a single solution build plan.
+     *
+     * @param exercise The exercise to migrate
+     * @param errorMap The error map to write errors to
+     */
+    private void migrateStudentBuildPlan(ProgrammingExercise exercise, ProgrammingExerciseStudentParticipation studentParticipation,
+            List<AuxiliaryRepository> auxiliaryRepositories, Map<ProgrammingExercise, Boolean> errorMap) {
+        VcsRepositoryUrl repositoryUrl;
+        try {
+            repositoryUrl = new VcsRepositoryUrl(urlService.getPlainUrlFromRepositoryUrl(studentParticipation.getVcsRepositoryUrl()));
+        }
+        catch (URISyntaxException e) {
+            log.warn("Failed to convert git url {} for studentParticipationId {} exerciseId {} with buildPlanId {}, will abort migration for this Participation",
+                    studentParticipation.getVcsRepositoryUrl(), studentParticipation.getId(), exercise.getId(), studentParticipation.getBuildPlanId(), e);
+            errorMap.put(exercise, false);
+            return;
+        }
+        try {
+            ciMigrationService.orElseThrow().overrideBuildPlanNotification(exercise.getProjectKey(), studentParticipation.getBuildPlanId(), repositoryUrl);
+        }
+        catch (VersionControlException | ContinuousIntegrationException | NoSuchElementException | RestClientException e) {
+            log.warn("Failed to migrate solution build plan for studentParticipationId {} with buildPlanId {} of exerciseId {} ", studentParticipation.getId(),
+                    studentParticipation.getBuildPlanId(), exercise.getId(), e);
+            errorMap.put(exercise, false);
+        }
+        try {
+            ciMigrationService.orElseThrow().deleteBuildTriggers(exercise.getProjectKey(), studentParticipation.getBuildPlanId(), repositoryUrl);
```

Comment:
IMO there should be a check so that we don't even attempt to delete the solution build triggers if the migration failed before. (same for similar cases)